api/src/main/java/com/epam/esm/service/validation/impl/CertificateValidatorImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.epam.esm.service.validation.impl;
+
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.service.validation.Validator;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+
+@Component
+public class CertificateValidatorImpl implements Validator<CertificateDto> {
```

Comment:
NIT: Spring has it's own validation infrastructure for that. See https://docs.spring.io/spring-framework/docs/3.2.x/spring-framework-reference/html/validation.html .